lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -332,9 +336,13 @@ pub fn get_match_checkers(index: &FieldIndex, cond_match: Match) -> Option<Condi
             (AnyVariants::Keywords(list), FieldIndex::KeywordIndex(index)) => {
                 Some(Box::new(move |point_id: PointOffsetType| {
                     index.get_values(point_id).map_or(false, |values| {
-                        values
-                            .iter()
-                            .any(|k| !list.iter().any(|s| s.as_str() == k.as_ref()))
+                        if list.len() < 13 {
```

Comment:
We should put this magic number in a constant along with a comment to explain where it comes from

Reply:
+1